src/python_testing/TC_DISHM_3_2.py

```diff
@@ -0,0 +1,122 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_DISHM_3_2(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.DishwasherMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def send_change_to_mode_cmd(self, newMode) -> Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse:
+        ret = await self.send_single_cmd(cmd=Clusters.Objects.DishwasherMode.Commands.ChangeToMode(newMode=newMode), endpoint=self.endpoint)
+        asserts.assert_true(type_matches(ret, Clusters.Objects.DishwasherMode.Commands.ChangeToModeResponse),
+                            "Unexpected return type for ChangeToMode")
+        return ret
+
+    async def write_start_up_mode(self, newMode):
+        ret = await self.default_controller.WriteAttribute(self.dut_node_id, [(self.endpoint, Clusters.DishwasherMode.Attributes.StartUpMode(newMode))])
+        asserts.assert_equal(ret[0].Status, Status.Success, "Writing to StartUpMode failed")
+
+    @async_test_body
+    async def test_TC_DISHM_3_2(self):
+
+        self.endpoint = self.user_params.get("endpoint", 1)
+
+        asserts.assert_true(self.check_pics("DISHM.S.A0000"), "DISHM.S.A0000 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.A0001"), "DISHM.S.A0001 must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C00.Rsp"), "DISHM.S.C00.Rsp must be supported")
+        asserts.assert_true(self.check_pics("DISHM.S.C01.Tx"), "DISHM.S.C01.Tx must be supported")
+
+        if not self.check_pics("DISHM.S.A0002"):
```

Comment:
Please also add code to check the preconditions for this test:

!DISHM.S.F00(DEPONOFF) | OnOff cluster’s StartUpOnOff attribute is NULL | StartUpOnOff is 0 | OnMode is NULL

I worked on this, for the RVC clusters and I just realized I didn't commit the changes! Here's what my code looked like. Please test it before adding it to this file ...


```
        depOnOffKey = "RVCCLEANM.S.F00"
        asserts.assert_true(depOnOffKey in self.matter_test_config.pics, "%s must be provided" % (depOnOffKey))

        ret = await self.check_preconditions(self.endpoint)
        asserts.assert_true(ret, "invalid preconditions - StartUpMode overridden by OnMode")


```
And this is the method that checks those preconditions:

 ```
  async def check_preconditions(self, endpoint):
        # check whether the StartUpMode will be overridden by the OnMode attribute

        if self.check_pics("RVCCLEANM.S.F00") == False:
            return True

        logging.info("RVCCLEANM.S.F00: 1")

        cluster = Clusters.Objects.OnOff
        attr = Clusters.OnOff.Attributes.StartUpOnOff
        startUpOnOff = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr)
        logging.info("StartUpOnOff: %s" % (startUpOnOff))
        if startUpOnOff == NullValue or startUpOnOff == 0:
            return True

        cluster = Clusters.Objects.RvcCleanMode
        attr = Clusters.RvcCleanMode.Attributes.OnMode
        onMode = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr)
        logging.info("OnMode: %s" % (onMode))
        if onMode == NullValue:
            return True

        return False



```